### PR TITLE
fix: use inappbrowser to handle add payment cards

### PR DESCRIPTION
**Pull request by reidzeibel**
closes https://github.com/AtB-AS/kundevendt/issues/18722

alternative of https://github.com/AtB-AS/mittatb-app/pull/4684

Now uses `inappbrowser`, which also works for Vipps login, and this removes the add payment screen since we no longer need to handle the web loading on our code.

### CODE REVIEW

**Comment by reidzeibel** [comment]
Weird, why do the test fails with message : `SyntaxError: Cannot use import statement outside a module`?

File: src/ticketing/api.ts [hunk 1]
```diff
@@ -57,12 +58,19 @@ export async function listRecurringPayments(): Promise<RecurringPayment[]> {
...
-  return response.data;
+  ).then(async (response) => {
+    await InAppBrowser.open(
+      response.data.terminal_url,
+      // Param showInRecents is needed so the InAppBrowser doesn't get closed when the app goes to background
+      // hence user is again navigated back to browser after finishing the Nets flow,
+      // and then can complete the authentication process successfully
+      {showInRecents: true},
+    );
+  });
```
Comment (gorandalum) [inline src/ticketing/api.ts:73]: I think we don't want to trigger opening of the in-app-browser in the api module.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx [hunk 2]
```diff
@@ -89,11 +90,11 @@ export const Profile_PaymentOptionsScreen = ({
       addPaymentMethodCallbackHandler,
     );
     return () => eventSubscription.remove();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [appStatus, authorizeRecurringPayment, cancelRecurringPayment]);
```
Comment (gorandalum) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx:93]: Do we need `appStatus` here?

File: src/ticketing/api.ts (hunk 1 above)
Comment (reidzeibel) [reply src/ticketing/api.ts:73]: I see, I only followed the Vipps Login flow on `src/api/vipps-login/api.ts`, I'll update it

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx (hunk 2 above)
Comment (reidzeibel) [reply src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx]: again, I followed the vipps login example, and we can remove it here, tested on dev build.

File: src/ticketing/api.ts (hunk 1 above)
Comment (reidzeibel) [reply src/ticketing/api.ts]: refactored

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx (hunk 2 above)
Comment (reidzeibel) [reply src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx]: refactored

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx [hunk 3]
```diff
@@ -90,11 +87,19 @@ export const Profile_PaymentOptionsScreen = () => {
...
-    await addPaymentMethod(redirectUrl);
+    await addPaymentMethod(redirectUrl).then(async (response) => {
+      await InAppBrowser.open(
+        response.data.terminal_url,
+        // Param showInRecents is needed so the InAppBrowser doesn't get closed when the app goes to background
+        // hence user is again navigated back to browser after finishing the Nets flow,
+        // and then can complete the authentication process successfully
+        {showInRecents: true},
+      );
+    });
```
Comment (gorandalum) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx]: This mix of `await` and `then`, with a nested `await` caught my eye. It isn't doing anything really wrong, but looks off 😅 What about just 
```ts
const response = await addPaymentMethod(redirectUrl);
await InAppBrowser.open(response.data.terminal_url, ...);
```

**Review by gorandalum — APPROVED** [review]

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx (hunk 3 above)
Comment (reidzeibel) [reply src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx]: Now that you mentioned it, it does look off... refactored

**Pull request merged**